### PR TITLE
Use group-id as part of project matching criteria

### PR DESCRIPTION
**Pull request by sashton**
Fix for #1

### CODE REVIEW

File: src/leiningen/checkout/ln.clj [hunk 1]
```diff
@@ -67,12 +67,17 @@
...
+  (fn [m] (map m ks)))
+
 (defn ln
   "[pattern]: Link project(s) into checkouts. If PATTERN is specified, link all projects matching `.*PATTERN.*`."
   [{:keys [name dependencies] {:keys [search-roots]} :checkout :as project} & [pattern]]
-  (let [dependency-names                        (into #{} (map (comp :artifact-id project/dependency-map) dependencies))
+  (let [dependency-coords                       (into #{} (map (comp (extract-vals [:group-id :artifact-id]) project/dependency-map) dependencies))
         {:keys [projects search-roots]}         (apply checkout-candidates project search-roots)
-        candidates-for-checkout                 (filter (comp dependency-names :name) projects)
+        candidates-for-checkout                 (filter (comp dependency-coords (extract-vals [:group :name])) projects)
```
Comment (timvisher) [inline src/leiningen/checkout/ln.clj]: Lest my eyes deceive me you're actually looking for `juxt`

```
user=> (map (juxt :a :b) [{:a 1 :b 2 :c 3} {:c 1 :b 3 :a 4}])
([1 2] [4 3])
```

**Comment by timvisher** [comment]
Your changes generally look good (in this and the other :pr:) but I can't actually get this (`master`) to run locally. Once I get that working I'll try to merge these in (assuming you respond to my feedback by then).

Thanks!

File: src/leiningen/checkout/ln.clj (hunk 1 above)
Comment (sashton) [inline src/leiningen/checkout/ln.clj]: Thanks for catching that. I love `juxt` and I was just using it elsewhere last week, but I didn't make the connection between it and the function I had added here. I've removed my function accordingly.

**Comment by sashton** [comment]
Hi @timvisher, how does this one look?

**Pull request merged**